streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceivedBlockTracker.scala

```diff
@@ -212,7 +213,7 @@ private[streaming] class ReceivedBlockTracker(
       writeAheadLog.readAll().asScala.foreach { byteBuffer =>
         logTrace("Recovering record " + byteBuffer)
         Utils.deserialize[ReceivedBlockTrackerLogEvent](
-          byteBuffer.array, Thread.currentThread().getContextClassLoader) match {
+          JavaUtils.bufferToArray(byteBuffer), Thread.currentThread().getContextClassLoader) match {
```

Comment:
Might be worth it to have a version of `Utils.deserialize` that takes a `ByteBuffer`, but not required for this change.